lib/vanagon/driver.rb

```diff
@@ -46,6 +48,24 @@ def load_engine(engine_type, platform, target)
       load_engine_object(engine_type, platform, target)
     end
 
+    # Should we be returning scheduled build hosts instead of 'hardware' engine
+    # hosts?
+    #
+    # REMIND: In the future we will likely allow configurations to specify
+    #   a specific engine. For now, we are introducing this to allow
+    #   transition for just which could use a "scheduled" engine.
+    def scheduled_hosts?(platform)
+      platform.engine && schedule_hardware_allocation?
+    end
+
+    # Should we allow the allocation of hosts via an external scheduler?
+    #
+    # REMIND: This is a feature flag method, triggered by an ENV setting
+    def schedule_hardware_allocation?
+      ENV['VANAGON_SCHEDULED_HARDWARE'] &&
+        !ENV['VANAGON_SCHEDULED_HARDWARE'].empty?
+    end
```

Comment:
Looking a bit closer, the cmdline options parsing for `build_host_info` passes in the value of the `--engine` flag, which ends up in our [`load_engine`](https://github.com/puppetlabs/vanagon/pull/417/files#diff-6d5e9c9d5b499a9e8d56ef00434459d9R36) method. Looks like that method just overwrites the `engine` value (before this PR).

Seems like we'd want to actually use that value if it's set, and **otherwise** go through the logic of setting a reasonable value.